File: src/main/java/com/dh/ondot/schedule/api/response/SettingAlarmResponse.java
package com.dh.ondot.schedule.api.response;

import com.dh.ondot.schedule.domain.Alarm;

public record SettingAlarmResponse(
        AlarmDto preparationAlarm,
        AlarmDto departureAlarm
) {
    public static SettingAlarmResponse from(Alarm preparation, Alarm departure) {
        return new SettingAlarmResponse(
                AlarmDto.of(preparation),
                AlarmDto.of(departure)
        );
    }
}

File: src/main/kotlin/com/dh/ondot/schedule/api/response/SettingAlarmResponse.kt
package com.dh.ondot.schedule.api.response

import com.dh.ondot.schedule.domain.Alarm

data class SettingAlarmResponse(
    val preparationAlarm: AlarmDto,
    val departureAlarm: AlarmDto,
) {
    companion object {
        @JvmStatic
        fun from(preparation: Alarm, departure: Alarm): SettingAlarmResponse {
            return SettingAlarmResponse(
                preparationAlarm = AlarmDto.of(preparation),
                departureAlarm = AlarmDto.of(departure),
            )
        }
    }
}
